common/common/src/main/java/io/helidon/common/SerializationConfig.java

```diff
@@ -0,0 +1,624 @@
+/*
+ * Copyright (c) 2021 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.common;
+
+import java.io.IOException;
+import java.io.ObjectInputFilter;
+import java.net.URL;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Properties;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+/**
+ * Support for JEP 290 - deserialization filtering.
+ * Configuration options mentioned below will differ in Helidon 3.0.0, the following table lists the options:
+ * <p>
+ * <table class="config">
+ * <caption>Configuration Options</caption>
+ * <tr>
+ *     <th>system property</th>
+ *     <th>2.x default</th>
+ *     <th>3.x default</th>
+ *     <th>description</th>
+ * </tr>
+ * <tr>
+ *     <td>{@link #PROP_WRONG_CONFIG_ACTION}</td>
+ *     <td>{@code warn} - {@link io.helidon.common.SerializationConfig.Action#WARN}</td>
+ *     <td>{@code fail} - {@link io.helidon.common.SerializationConfig.Action#FAIL}</td>
+ *     <td>What to do if an existing global deserialization filter exists without a global blacklist.</td>
```

Comment:
Maybe use allow-list and reject-list (and allow and reject) to align with https://docs.oracle.com/en/java/javase/11/core/serialization-filtering1.html